packages/nomad/api.hcl

```diff
@@ -143,6 +143,16 @@ job "api" {
           "--port", "${var.api_port_number}",
         ]
       }
+
+      template {
+	env = true
+	destination = "local/env.txt"
+	data = <<EOF
+{{ range nomadService "redis" }}
+REDIS_URL=redis://{{ .Address }}:{{ . Port }}
+{{ end }}
+EOF
+      }
```

Comment:
you can use `redis.service.consul` to dynamically resolve the IP. I am not sure if redeploy of redis wouldn't cause redeploy of api in this case